pyfr/plugins/writer.py

```diff
@@ -58,6 +58,18 @@ def __call__(self, intg):
         else:
             metadata = None
 
+        # Fetch data from other plugins and add it to metadata with ad-hoc keys
+        for csh in intg.completed_step_handlers:
+            try:
+                pdata = csh.serialise(intg)
+                prefix = intg.get_plugin_data_prefix(csh.name, csh.suffix)
```

Comment:
This can be moved out of the try as it should always succeed.  Also may want to have the method take `csh` as an argument and let it figure out what info it wants to use.

Reply:
This belongs to PR242. It is not going to always succeed as `csh` may not be a plugin. And yep, I changed the method to take `csh` as an argument.

Reply:
I'd be tempted to force all plugins to be subclasses of BasePlugin.  I think the only one which isn't is the progress bar, but this is easy enough to update.